applications/GeoMechanicsApplication/tests/test_set_multiple_moving_load_process.py

```diff
@@ -24,11 +66,51 @@ def setUp(self):
                     "origin"                    : [0,0,0]
                 }
                 """)
+        # Add nodal solution space
+        self.mp.AddNodalSolutionStepVariable(KratosMultiphysics.DISPLACEMENT)
+        self.strategy = self.setup_strategy(self.cmp)
+
+    def create_two_nodes_condition(self, reverse_nodes=False):
+        # create nodes
+        self.mp.CreateNewNode(1, 0.0, 0.0, 0.0)
+        self.mp.CreateNewNode(2, 1.0, 0.0, 0.0)
+        self.set_nodal_displacement_dof()
+
+        # create condition
+        if reverse_nodes:
+            self.mp.CreateNewCondition("MovingLoadCondition2D2N", 1, [2, 1], self.mp.GetProperties()[1])
+        else:
+            self.mp.CreateNewCondition("MovingLoadCondition2D2N", 1, [1, 2], self.mp.GetProperties()[1])
+
+    def create_three_nodes_condition(self, reverse_nodes=False):
+        # create nodes
+        self.mp.CreateNewNode(1, 0.0, 0.0, 0.0)
+        self.mp.CreateNewNode(2, 1.0, 0.0, 0.0)
+        self.mp.CreateNewNode(3, 2.0, 0.0, 0.0)
+        self.set_nodal_displacement_dof()
+
+        # create condition
+        if reverse_nodes:
+            self.mp.CreateNewCondition("MovingLoadCondition2D2N", 1, [3, 2], self.mp.GetProperties()[1])
+            self.mp.CreateNewCondition("MovingLoadCondition2D2N", 2, [2, 1], self.mp.GetProperties()[1])
+        else:
+            self.mp.CreateNewCondition("MovingLoadCondition2D2N", 1, [1, 2], self.mp.GetProperties()[1])
+            self.mp.CreateNewCondition("MovingLoadCondition2D2N", 2, [2, 3], self.mp.GetProperties()[1])
+
+    def initialize_test(self, process):
+        process.ExecuteInitialize()
+        process.ExecuteInitializeSolutionStep()
+        self.strategy.InitializeSolutionStep()
+
+    def next_solution_step(self, process):
+        process.ExecuteFinalizeSolutionStep()
```

Comment:
I see it was already the case before, but why do we first execute finalize and afterwards the initialize?

Reply:
Basically because next_solution_step, is between steps, whereas initialize_test is the first.